views/student-analytics.pug

```diff
@@ -0,0 +1,22 @@
+.row#student-analytics-card
+  if (isAdmin())
```

Comment:
Why are you referencing admin rights on student-analytics page? This should be separated in analytics.pug.

Reply:
The reason for this is that this file is a shared  component between both admin and student. It is not purely for the student.

Reply:
Can we not isolate the two?

Reply:
Better way is to reference this in the admin analytics page

Reply:
Isolation is not needed since this is a shared component. It would add more files and clutter the codebase

Reply:
Another point is that if we split it off the code will will have the exact same file with with a different name and without the if statement. This is duplicate code. We will also have to add identical paths in the backend and have a different name for each path, again code duplication.

Reply:
@mehtadarshan1 this is the same way it is in question preview where if the prof is viewing a preview of the question, "if preview", then it adds/removes an extra element. I think this is good.

Reply:
Then what is analytics.pug used for? I thought that is for admin

Reply:
`analytics.pug` unifies the 2 types of analytics (student and class) using the materialized functionality `cards` and `tabs`. It also imports all the `CSS` and `JS` files and libraries required for both analytics to run, resulting in no code duplication

Reply:
I was thinking more for analytics security to not mix up admin and students.

Reply:
There are no security issues, admin needs to see both sides. Students will only have access to the student analytics as admin analytics is not an option that they can see

Reply:
sounds good (Y)

Reply:
🖖👍☺️